neo/Consensus/ConsensusService.cs

```diff
@@ -134,6 +115,41 @@ public void Dispose()
             }
         }
 
+        private void FillContext()
+        {
+            List<Transaction> transactions = LocalNode.GetMemoryPool().Where(p => CheckPolicy(p)).ToList();
+            if (transactions.Count >= Settings.Default.MaxTransactionsPerBlock)
+                transactions = transactions.OrderByDescending(p => p.NetworkFee / p.Size).Take(Settings.Default.MaxTransactionsPerBlock - 1).ToList();
+            Fixed8 amount_netfee = Block.CalculateNetFee(transactions);
+            TransactionOutput[] outputs = amount_netfee == Fixed8.Zero ? new TransactionOutput[0] : new[] { new TransactionOutput
+            {
+                AssetId = Blockchain.UtilityToken.Hash,
+                Value = amount_netfee,
+                ScriptHash = wallet.GetChangeAddress()
+            } };
+            while (true)
+            {
+                ulong nonce = GetNonce();
+                MinerTransaction tx = new MinerTransaction
+                {
+                    Nonce = (uint)(nonce % (uint.MaxValue + 1ul)),
+                    Attributes = new TransactionAttribute[0],
```

Comment:
Maybe we can force the unique hash adding the block index

`                Attributes = new TransactionAttribute[] { new TransactionAttribute() { Data = BitConverter.GetBytes(context.BlockIndex), Usage = TransactionAttributeUsage.Remark } },
`

Reply:
Why not just using `Nonce`?